src/modules/stripe/stripe.service.ts

```diff
@@ -0,0 +1,159 @@
+import {
+  BadRequestException,
+  ConflictException,
+  Inject,
+  Injectable,
+  InternalServerErrorException,
+  NotFoundException,
+} from '@nestjs/common';
+import { ConfigService } from '@nestjs/config';
+
+import { Errors } from 'src/common/errors';
+import { CANADA_POST_LOGO } from 'src/config';
+import { mockOrders } from 'src/mocks/mock-orders';
+import { CartService } from 'src/modules/cart/cart.service';
+import { ProductsService } from 'src/modules/products/products.service';
+import Stripe from 'stripe';
+
+import { UsersService } from '../users/users.service';
+
+import { PaymentDto } from './dto/payment.dto';
+import { StripeModuleOptions } from './stripe.interfaces';
+import { MODULE_OPTIONS_TOKEN } from './stripe.module-definition';
+
+const minDifference = 0.01;
+const centsInDollar = 100;
+
+@Injectable()
+export class StripeService {
+  public readonly StripeApi: Stripe;
+  constructor(
+    @Inject(MODULE_OPTIONS_TOKEN) private options: StripeModuleOptions,
+    private usersServise: UsersService,
+    private productsService: ProductsService,
+    private configService: ConfigService,
+    private cartService: CartService,
+  ) {
+    this.StripeApi = new Stripe(this.options.apiKey, this.options.options);
+  }
+
+  async createCheckoutSession(
+    customerId: string,
+    payment: PaymentDto,
+  ): Promise<Stripe.Response<Stripe.Checkout.Session>> {
+    const { productIds, total, shippingPrice } = payment;
+
+    try {
+      const user = await this.usersServise.getUserById(customerId);
+
+      if (!user) {
+        throw new NotFoundException(Errors.USER_NOT_FOUND);
+      }
+
+      const products = await this.productsService.findByIds(productIds);
+
+      if (products.length < productIds.length) {
+        throw new NotFoundException(Errors.SOME_PRODUCTS_NOT_FOUND);
+      }
+
+      const productsTotalPrice = products.reduce(
+        (sum, product) => sum + Number(product.price),
+        0,
+      );
+
+      if (
+        Math.abs(productsTotalPrice + shippingPrice - total) > minDifference
+      ) {
+        throw new ConflictException(Errors.PRICES_DO_NOT_MATCH);
```

Comment:
how you will handle it on frontend?

Reply:
The user will see a toast notification indicating that the prices of the items have changed. Then, the cart will reload, showing the new prices, and the user will be able to confirm the payment.